blob/s3blob/s3blob.go

```diff
@@ -144,24 +143,58 @@ type URLOpener struct {
 	Options Options
 }
 
+const (
+	sseTypeParamKey  = "ssetype"
+	kmsKeyIdParamKey = "kmskeyid"
+)
+
+func IsValidServerSideEncryptionType(value string) bool {
+	// Surely there is a better way to get the values
+	for _, v := range typesv2.ServerSideEncryptionAes256.Values() {
+		if string(v) == value {
```

Comment:
Optional: Make the comparison case insensitive using strings.ToLower on both sides.